internal/database/migrations/1_initial_schema.up.sql

```diff
@@ -0,0 +1,171 @@
+    provider VARCHAR(255) NOT NULL,
+    provider_id VARCHAR(255) NOT NULL,
+    created_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+    updated_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+    UNIQUE(provider, provider_id)
+);
+
+CREATE TABLE refresh_tokens (
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    user_id UUID NOT NULL REFERENCES users(id) ON DELETE CASCADE,
+    expiration_date TIMESTAMPTZ NOT NULL,
+    is_active BOOLEAN DEFAULT TRUE,
+    created_at TIMESTAMPTZ DEFAULT now(),
+    updated_at TIMESTAMPTZ DEFAULT now()
+);
+
+-- Organizational And Unit Tables
+
+CREATE TYPE unit_type AS ENUM ('organization', 'unit');
+
+CREATE TABLE IF NOT EXISTS units (
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    org_id UUID REFERENCES units(id),
+    parent_id UUID REFERENCES units(id) ON DELETE SET NULL,
+    type unit_type NOT NULL DEFAULT 'unit',
+    name VARCHAR(255),
+    description VARCHAR(255),
+    metadata JSONB,
+    created_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+    updated_at TIMESTAMPTZ NOT NULL DEFAULT now()
+);
+
+CREATE INDEX idx_units_parent_id ON units(parent_id);
+
+CREATE TABLE IF NOT EXISTS unit_members (
+    unit_id UUID REFERENCES units(id) ON DELETE CASCADE,
+    member_id UUID,
+    PRIMARY KEY (unit_id, member_id)
+);
+
+CREATE TYPE db_strategy AS ENUM ('shared', 'isolated');
+
+CREATE TABLE IF NOT EXISTS tenants (
+    id          UUID PRIMARY KEY REFERENCES units (id) ON DELETE CASCADE,
+    slug        TEXT UNIQUE NOT NULL,
+    db_strategy db_strategy NOT NULL,
+    owner_id    UUID REFERENCES users (id) ON DELETE SET NULL
+);
+
+-- Forms Tables
+
+CREATE TYPE status AS ENUM(
+    'draft',
+    'published'
+);
+
+CREATE TYPE question_type AS ENUM(
+    'short_text',
+    'long_text',
+    'single_choice',
+    'multiple_choice',
+    'date'
+);
+
+CREATE TABLE IF NOT EXISTS forms (
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    title TEXT NOT NULL,
+    description TEXT,
+    preview_message TEXT,
+    status status NOT NULL DEFAULT 'draft',
+    deadline TIMESTAMPTZ,
+    unit_id UUID REFERENCES units(id) ON DELETE CASCADE,
+    last_editor UUID NOT NULL REFERENCES users(id) ON DELETE CASCADE,
+    created_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+    updated_at TIMESTAMPTZ NOT NULL DEFAULT now()
+);
+
+CREATE TABLE IF NOT EXISTS questions(
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    form_id UUID NOT NULL REFERENCES forms(id) ON DELETE CASCADE,
+    required BOOLEAN NOT NULL,
+    type question_type NOT NULL,
+    title TEXT,
+    description TEXT,
+    metadata JSONB DEFAULT '{}'::JSONB,
+    "order" INTEGER NOT NULL,
+    created_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+    updated_at TIMESTAMPTZ NOT NULL DEFAULT now()
+);
+
+CREATE TABLE IF NOT EXISTS form_responses (
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    form_id UUID NOT NULL REFERENCES forms(id) ON DELETE CASCADE,
+    submitted_by UUID NOT NULL REFERENCES users(id) ON DELETE CASCADE,
```

Comment:
Consider remove `ON DELETE CASCADE` since it is not practical to remove a response when a user it removed.
```suggestion
    submitted_by UUID NOT NULL REFERENCES users(id),
```